syncer/syncer.go

```diff
@@ -681,13 +681,24 @@ func (s *Syncer) Run(ctx context.Context) error {
 
 // Close closes the Syncer's net.Listener.
 func (s *Syncer) Close() error {
+	s.mu.Lock()
```

Comment:
Is it crazy to derive a cancel from the context we get passed in run and call that here? That would make this a lot more robust.

Reply:
Absolutely crazy to pass contexts around outside of their scope

Reply:
I agree that passing contexts around outside of their "context" is not great. So I'd leave it the way it is for now and consider a follow up with a dedicated context or thread group for the `Syncer`.